odl/solvers/nonsmooth/adupdates.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2014-2017 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Alternating dual (AD) upgrade algorithm studied by McGaffin and Fessler.
+
+The alternating dual upgrade method is reported to be a flexible method which
+allows an acceleration by permitting efficient calculations on a graphics card.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+
+import numpy as np
+
+__all__ = ('adupdates',)
+
+
+def adupdates(funcs, ops, x, stepsize, majs, niter, random=False,
+              callback=None, callback_loop='outer'):
```

Comment:
I suggest that we use the same signature (where possible) as for the Douglas-Rachford method, i.e. `g` for functionals that are composed, `L` for the operators, etc.
See also #1197

Reply:
I am currently doing this and I noticed that there is the parameter kwargs, which is not documented in the docstring of the Douglas–Rachford method. In the iterative methods, this parameter is not present at all. Which variant shall I implement here?

Reply:
We don't really have a rule on that, but if you feel that the signature gets a bit too long with all the optional parameters, stuff the less common, less important or more advanced ones into `kwargs`. In this case, I'd say it's fine without `kwargs` since you have only one optional parameter `random` (I don't count `callback`s).